### PR TITLE
feat(flutter_availability_data_interface): define the models and interface for availabilities

### PR DESCRIPTION
**Pull request by FlutterJoey**

### CODE REVIEW

File: packages/flutter_availability_data_interface/lib/src/models/availability.dart [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      );
+}
+
+/// A model defining the structure of a break within an [AvailabilityModel]
+class AvailabilityBreakModel {
+  /// Create a new AvailabilityBreakModel
+  AvailabilityBreakModel({
+    required this.startTime,
+    required this.endTime,
+    required this.duration,
```
Comment (PureTryOut) [inline packages/flutter_availability_data_interface/lib/src/models/availability.dart]: Any reason you're making it required even though it's nullable?

File: packages/flutter_availability_data_interface/lib/src/data_interface.dart [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+/// story to its persistance solution.
+///
+/// This class needs to be implemented for your use case, for example a REST API
+/// or a Firebase database.
+abstract interface class AvailabilityDataInterface {
+  /// Retrieves a list of availabilities for the given [userId].
+  ///
+  /// Whether this is a one time value or a continuous stream of values is up to
+  /// the implementation.
+  Stream<List<AvailabilityModel>> getAvailabilitieForUser(String userId);
```
Comment (PureTryOut) [inline packages/flutter_availability_data_interface/lib/src/data_interface.dart]:
```suggestion
  Stream<List<AvailabilityModel>> getAvailabilityForUser(String userId);
```

Comment (freekvandeven) [inline packages/flutter_availability_data_interface/lib/src/data_interface.dart:13]: Availability

**Review by PureTryOut — CHANGES_REQUESTED** [review]

File: packages/flutter_availability_data_interface/lib/src/data_interface.dart [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  /// Retrieves a specific availability for the given
+  /// [userId] and [availabilityId]
+  Stream<AvailabilityModel> getAvailabilityForUserById(
+    String userId,
+    String availabilityId,
+  );
+
+  /// Deletes the availability with id [availabilityId] for this user
```
Comment (freekvandeven) [inline packages/flutter_availability_data_interface/lib/src/data_interface.dart:22]: Could you do the same style of comment as the one above

File: packages/flutter_availability_data_interface/lib/src/models/availability.dart [hunk 4]
```diff
@@ -0,0 +1,92 @@
+/// A model defining the data structure for an availability
+class AvailabilityModel {
+  /// Creates a new availability
+  AvailabilityModel({
```
Comment (freekvandeven) [inline packages/flutter_availability_data_interface/lib/src/models/availability.dart:4]: const

File: packages/flutter_availability_data_interface/lib/src/models/availability.dart [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        startDate: startDate ?? this.startDate,
+        endDate: endDate ?? this.endDate,
+        breaks: breaks ?? this.breaks,
+      );
+}
+
+/// A model defining the structure of a break within an [AvailabilityModel]
+class AvailabilityBreakModel {
+  /// Create a new AvailabilityBreakModel
+  AvailabilityBreakModel({
```
Comment (freekvandeven) [inline packages/flutter_availability_data_interface/lib/src/models/availability.dart:52]: const

**Review by freekvandeven — CHANGES_REQUESTED** [review]

File: packages/flutter_availability_data_interface/lib/src/models/availability.dart (hunk 1 above)
Comment (FlutterJoey) [reply packages/flutter_availability_data_interface/lib/src/models/availability.dart]: I forgot some of the information on the model, will append it now

**Review by freekvandeven — APPROVED** [review]

**Review by PureTryOut — APPROVED** [review]

**Pull request merged**